spec/java/util/LinkedList.lsl

```diff
@@ -0,0 +1,666 @@
+       	 	return index >= 0 && index < size;
+    	}
+	
+	
+	@Private
+	sub isPositionIndex(index: int): boolean {
+        	return index >= 0 && index <= size;
+    	}
+	
+	
+	@Private
+	sub checkPositionIndex(index: int) {
+        	if (!isPositionIndex(index))
+		{
+            		//Работает ли такая конкатенация строк и можно ли внутри ифа  объявить локальную переменную
+			var message: string =  "Index: "+index+", Size: "+size;
+			action THROW_NEW('java.util._IndexOutOfBoundsException', [message]);
+		}
+    	}
+	
+	
+	fun get (index: int): Object {
+		checkElementIndex(index);
+		result = action LIST_GET(storage, index);
+	}
+
+	
+	fun set (index: int, element: Object): Object {
+		checkElementIndex(index);
+		action LIST_SET(storage, index, element);
+		result = action LIST_GET(storage, index);
+	}
+	
+	
+	fun add(index: int, element: Object): void {
+		checkPositionIndex(index);
+		linkAny(index, element);
+	}
+	
+	
+	fun remove (index: int): Object {
+		checkElementIndex(index);
+		result = unlinkAny(index);
+	}
+	
+	
+	fun indexOf(o: Object): int {
+		result = action LIST_FIND(storage,o, 0, size, 1);
+	}
+	
+	
+	fun lastIndexOf(o: Object): int {
+		result = action LIST_FIND(storage,o, size, 0, -1);
+	}
+	
+	
+	fun peek(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+			result = action LIST_GET(storage, 0);
+		}
+	}
+	
+	
+	fun element (): Object {
+		result = getFirst();	
+	}
+	
+	
+	fun poll(): Object {
+		if(size==0)
+		{
+			result = null;
+		}
+		else 
+		{
+			result = unlinkAny(0);
+		}
+	}
+	
+	
+	fun remove(): Object {
+		result = removeFirst();
+	}
+	
+	
+	fun offer(e: Object): boolean {
+	 	add(e);
+		result = true;
+	}
+
+	
+	fun offerFirst(e: Object): boolean {
+		addFirst(e);
+		result = true;
+	}
+
+
+	fun offerLast(e: Object): boolean {
+		addLast(e);
+		result = true;
+	}
+	
+	
+	fun peekFirst(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+			result = action LIST_GET(storage, 0);
+		}
+	}
+	
+	
+	fun peekLast(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+			result = action LIST_GET(storage, size-1);
+		}
+	}
+
+
+	fun pollFirst(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+			result = unlinkAny(0);
+		}
+	}
+	
+	
+	fun pollLast(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+ 			result = unlinkAny(size-1);
+		}
+	}
+	
+	
+	fun push(e: Object): void {
+		addFirst(e);
+	}
+
+
+	fun pop(): Object {
+		result = removeFirst();
+	}
+	
+	
+	fun removeFirstOccurrence(o: Object): boolean {
+		result = remove(o);
+	}
+	
+	
+	fun removeLastOccurrence(o: Object): boolean {
+		//I need think about this
+	}
+	
+	
+	//We need add type: typealias ArrayObject = array<Object>;
+	fun toArray(a: ArrayObject): ArrayObject {
+		result = action LIST_TO_ARRAY(storage, a);
+	}
+
+	
+	fun spliterator(): Spliterator {
+		result = new LLSpliterator(state=Initialized,
+		//This is right ? "parent=self"
+		parent=self,
```

Comment:
Please use additional spacing in case of complex object creation when multiple lines used for specifying variable states.

Reply:
Has been fixed.